### PR TITLE
Fix ci for windows and Intel ifx (#7)

### PR DESCRIPTION
**Pull request by jvdp1**
Fixes multiple issues with CI (Ubuntu + Intel and Windows)

Changes:

* Windows: GCC 13
* Ubuntu-latest: Intel LLVM (instead of Classic)
* Relax test in test_rawmoment due to change from Intel Classic to Intel LLVM

Note: CI for Windows works, but could be most likely improved (but how?)

### CODE REVIEW

**Comment by jvdp1** [comment]
This PR should be merged for enabling (at least) #747 #743

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -92,12 +92,78 @@ jobs:
...
+        cc: [icx]
+        cxx: [icpx]
+    env:
+      FC: ${{ matrix.fc }}
+      CC: ${{ matrix.cc }}
+      CXX: ${{ matrix.cxx }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
```
Comment (henilp105) [inline .github/workflows/CI.yml:111]: can we shift to the v4 , there had been some bugs in 'v2'.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -92,12 +92,78 @@ jobs:
...
+
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v1
```
Comment (henilp105) [inline .github/workflows/CI.yml:114]: can we shift to 'v5' .

**Review by henilp105 — APPROVED** [review]
Thanks @jvdp1 , Looks good to me. I have suggested some minor changes.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (jvdp1) [reply .github/workflows/CI.yml]: Thank you. I applied this change.

File: .github/workflows/CI.yml (hunk 2 above)
Comment (jvdp1) [reply .github/workflows/CI.yml]: Thank you. I applied this change.

**Review by awvwgk — APPROVED** [review]

**Comment by jvdp1** [comment]
There are 2 approvals. I will merge it to allow others to be merged too.

**Pull request merged**